package.json

```diff
@@ -1,17 +1,17 @@
 {
   "name": "indie-stack-template",
+  "type": "module",
   "private": true,
   "sideEffects": false,
   "scripts": {
-    "build": "remix build",
-    "dev": "remix dev -c \"npm run dev:serve\"",
-    "dev:serve": "binode --require ./mocks -- @remix-run/serve:remix-serve ./build/index.js",
+    "build": "remix vite:build",
+    "dev": "remix vite:dev",
     "format": "prettier --write .",
     "format:repo": "npm run format && npm run lint -- --fix",
     "lint": "eslint --cache --cache-location ./node_modules/.cache/eslint .",
     "setup": "prisma generate && prisma migrate deploy && prisma db seed",
-    "start": "remix-serve ./build/index.js",
-    "start:mocks": "binode --require ./mocks -- @remix-run/serve:remix-serve ./build/index.js",
+    "start": "remix-serve ./build/server/index.js",
+    "start:mocks": "binode --import ./mocks/index.js -- @remix-run/serve:remix-serve ./build/server/index.js",
     "test": "vitest",
     "test:e2e:dev": "start-server-and-test dev http://localhost:3000 \"npx cypress open\"",
     "pretest:e2e:run": "npm run build",
```

Comment:
facing issues while running `validate` as `npx cypress run` command is failing with an error `ReferenceError: exports is not defined in ES module scope`

Tried updating `tsconfig.json` by following https://github.com/cypress-io/cypress/issues/23552#issuecomment-1606338593 but couldn't make it work. Will check again

Reply:
I've migrated an Indie Stack application to vite.
I'd recommend first moving to ESM and then afterwards migrating to vite. Makes it easier to know where the error is from 😅 

Regarding your issue I ended up switching to `tsx` instead of `ts-node` as I couldn't get it to work with `ts-node`.
Using tsx I modified the file `cypress/support/commands.ts` according to this:
```diff
-cy.exec(
-  `npx ts-node -r tsconfig-paths/register ./cypress/support/create-user.ts "${email}"`,
-).then(({ stdout }) => {
-  const cookieValue = stdout
-    .replace(/.*<cookie>(?<cookieValue>.*)<\/cookie>.*/s, "$<cookieValue>")
-    .trim();
-  cy.setCookie("__session", cookieValue);
-});
+cy.exec(`tsx ./cypress/support/create-user.ts "${email}"`).then(
+  ({ stdout }) => {
+    const cookieValue = stdout
+      .replace(/.*<cookie>(?<cookieValue>.*)<\/cookie>.*/s, "$<cookieValue>")
+      .trim();
+    cy.setCookie("__session", cookieValue);
+  },
+);

-cy.exec(
-  `npx ts-node -r tsconfig-paths/register ./cypress/support/delete-user.ts "${email}"`,
-);
+cy.exec(`tsx ./cypress/support/delete-user.ts "${email}"`);
```

And `package.json`
```json
"scripts": {
    "build": "remix vite:build",
    "dev": "binode --import ./mocks/index.js -- @remix-run/dev:remix vite:dev",
    "format": "prettier --write .",
    "lint": "eslint --cache --cache-location ./node_modules/.cache/eslint .",
    "setup": "prisma generate && prisma migrate deploy && prisma db seed",
    "start": "remix-serve ./build/server/index.js",
    "start:mocks": "binode --import ./mocks/index.js -- @remix-run/serve:remix-serve ./build/server/index.js",
    "test": "vitest",
    "test:e2e:dev": "start-server-and-test dev http://localhost:3000 \"npx cypress open\"",
    "pretest:e2e:run": "npm run build",
    "test:e2e:run": "cross-env PORT=8811 start-server-and-test start:mocks http://localhost:8811 \"npx cypress run\"",
    "typecheck": "tsc && tsc -p cypress",
    "validate": "run-p \"test -- --run\" lint typecheck test:e2e:run",
    "prisma:studio": "prisma studio",
    "optimize:images": "tsx app/optimize-images.ts"
  },
```

Reply:
@Apsysikal, thanks for the input. I tried `tsx` and still getting `zsh:1: command not found: tsx`

Reply:
Could you confirm what shell are you using? I read there's some issue with zsh in particular with Cypress

https://github.com/cypress-io/cypress/issues/6081

Reply:
<img width="1269" alt="Screenshot 2024-05-04 at 7 34 37 PM" src="https://github.com/remix-run/indie-stack/assets/31823611/46a73a56-567f-4679-8d4c-c6ff210e6b09">

Reply:
I also got an error (different one) but fixed it on my end. I used bash. The changes made (should also probably update the comment) are:

`cypress/support/create-user.ts`

```diff
-// npx ts-node -r tsconfig-paths/register ./cypress/support/create-user.ts username@example.com,
+// npx tsx ./cypress/support/create-user.ts username@example.com,
```

`cypress/support/delete-user.ts`

```diff
-// npx ts-node -r tsconfig-paths/register ./cypress/support/delete-user.ts username@example.com,
+// npx tsx ./cypress/support/delete-user.ts username@example.com,

-import { PrismaClientKnownRequestError } from "@prisma/client/runtime";
+import { Prisma } from "@prisma/client";

-error instanceof PrismaClientKnownRequestError &&
+error instanceof Prisma.PrismaClientKnownRequestError &&
```

You should also add tsx to the `devDependencies`.

`package.json`

```diff
+"tsx": "^4.9.1",
```